pkg/kubelet/apis/config/scheme/scheme.go

```diff
@@ -35,6 +35,6 @@ func NewSchemeAndCodecs() (*runtime.Scheme, *serializer.CodecFactory, error) {
 	if err := kubeletconfigv1beta1.AddToScheme(scheme); err != nil {
 		return nil, nil, err
 	}
-	codecs := serializer.NewCodecFactory(scheme)
+	codecs := serializer.NewCodecFactory(scheme, serializer.EnableStrict)
```

Comment:
for non-alpha config file formats (like the v1beta1 format the kubelet has), I would only expect to strengthen validation on a version boundary (e.g. v1beta2, etc)

I'd suggest returning a strict codec, and a non-strict codec that only knows about v1beta1 (and would not get future versions registered into it), and doing the following where we load config:
```go
config, strictErr := ... decode with strictCodec ...
if strictErr != nil {
  lenientConfig, err := ... decode with lenientV1beta1Codec ...
  if err != nil {
    return strictErr
  }
  ... log warning with strict errors
  config = lenientConfig
}
```

once v1beta1 kubelet config is deprecated and removed, we can drop the lenient path

Reply:
I'd be ok with that, as a way of being extra cautious.

Reply:
I've added the lenient path

Reply:
@liggitt, did you mean returning two codecs from this function (that's how I interpreted your recommendation)?